tools/devops/automation/templates/packages/build.yml

```diff
@@ -0,0 +1,454 @@
+    IOS_PKG=$(find $(Build.SourcesDirectory)/package -type f -name "xamarin.ios-*" | xargs basename)
+    if [ -z "$IOS_PKG" ]; then
+      echo "Xamarin.iOS package not found."
+    else
+      IOS_PKG="$(Build.SourcesDirectory)/package/$IOS_PKG"
+      echo "##vso[task.setvariable variable=IOS_PKG;]$IOS_PKG"
+      echo "Xamarin.iOS package found at $IOS_PKG"
+    fi
+
+    MAC_PKG=$(find $(Build.SourcesDirectory)/package -type f -name "xamarin.mac-*" | xargs basename)
+    if [ -z "$MAC_PKG" ]; then
+      echo "Xamarin.Mac package not found."
+    else
+      MAC_PKG="$(Build.SourcesDirectory)/package/$MAC_PKG"
+      echo "##vso[task.setvariable variable=MAC_PKG;]$MAC_PKG"
+      echo "Xamarin.Mac package found at $MAC_PKG"
+    fi
+  name: packages
+  displayName: 'Build Packages'
+  condition: and(succeeded(), eq(variables['configuration.BuildPkgs'], 'True'))
+  timeoutInMinutes: 180
+
+# build nugets
+- bash:  $(Build.SourcesDirectory)/xamarin-macios/tools/devops/automation/scripts/bash/build-nugets.sh
+  displayName: 'Build Nugets'
+  condition: and(succeeded(), eq(variables['configuration.BuildNugets'], 'True'))
+  continueOnError: true # should not stop the build since is not official just yet.
+  timeoutInMinutes: 180
+
+- bash: $(Build.SourcesDirectory)/xamarin-macios/tools/devops/automation/scripts/bash/productsign.sh
+  env:
+    PRODUCTSIGN_KEYCHAIN_PASSWORD: $(xma-password)
+  displayName: 'Signing PR Build'
+  condition: and(succeeded(), eq(variables['configuration.SignPkgs'], 'True'), eq(variables['configuration.IsPr'], 'True'))
+
+  # Ensure virtualenv is on the PATH
+- template: set-path/v1.yml@templates
+  parameters:
+    prependToPath: '/Users/builder/Library/Python/2.7/bin'
+
+- bash: |
+    VIRTUAL_ENV_PATH=$(Build.SourcesDirectory)/venv
+    pip install virtualenv
+    virtualenv "$VIRTUAL_ENV_PATH" --system-site-packages
+    source "$VIRTUAL_ENV_PATH/bin/activate"
+    pip install python-magic
+
+    security unlock-keychain -p $PRODUCTSIGN_KEYCHAIN_PASSWORD builder.keychain
+    PACKAGES="$IOS_PKG $MAC_PKG"
+    echo "Packages found at $PACKAGES"
+
+    echo "$PACKAGES" | xargs python $(Build.SourcesDirectory)/release-scripts/sign_and_notarize.py -a "$APP_ID" -i "$INSTALL_ID" -u "$APPLE_ACCOUNT" -p "$APPLE_PASS" -t "$TEAM_ID" -d $(Build.SourcesDirectory)/package/notarized -e "$MAC_ENTITLEMENTS" -k "$KEYCHAIN"
+
+    deactivate
+    rm -Rf "$VIRTUAL_ENV_PATH"
+  env:
+    PRODUCTSIGN_KEYCHAIN_PASSWORD: $(OSX_KEYCHAIN_PASS)
+    MAC_ENTITLEMENTS: $(Build.SourcesDirectory)/xamarin-macios/mac-entitlements.plist
+    APP_ID: $(XamarinAppId)
+    INSTALL_ID: $(XamarinAppId)
+    APPLE_ACCOUNT: $(XamarinUserId)
+    APPLE_PASS: $(XamarinPassword)
+    TEAM_ID: $(TeamID)
+    KEYCHAIN: $(SigningKeychain)
+  name: notarize
+  displayName: 'Signing Release Build'
+  condition: and(succeeded(), eq(variables['configuration.SignPkgs'], 'True'), eq(variables['configuration.IsPr'], 'False'))
```

Comment:
Why are PR builds and CI builds signed differently?

Reply:
Honestly, I don't know I followed https://github.com/xamarin/xamarin-macios/blob/main/jenkins/Jenkinsfile#L568